ResearchKit/Common/ORKTaskViewController.m

```diff
@@ -1292,6 +1292,8 @@ - (ORKStepViewController *)viewControllerForStep:(ORKStep *)step {
     
     if (stepViewController.cancelButtonItem == nil && !stepViewController.step.saveOnArrival) {
         stepViewController.cancelButtonItem = [self defaultCancelButtonItem];
+    } else if (stepViewController.cancelButtonItem && stepViewController.step.saveOnArrival) {
```

Comment:
These same `if (stepViewController.cancelButtonItem == nil && !stepViewController.step.saveOnArrival)` blocks also appear in ORK1TaskViewController, is there a reason you're not adding these `else if` blocks there?

Reply:
Ah, this was an artifact from some early attempts to get rid of the cancel button until I realized it is created differently and behaves differently in RK2 than in RK1. I have now removed this and the tests still pass :)